include/AnalysisTools.h

```diff
@@ -293,6 +293,33 @@ RVecI FindMatching(const RVecLV& target_p4, const RVecLV& ref_p4,const float del
   return targetIndices;
 }
 
+RVecI FindMatching(const RVecB& pre_sel_target, const RVecB& pre_sel_ref, const RVecLV& target_p4, const RVecLV& ref_p4,const float deltaR_thr){
+  RVecI targetIndices(target_p4.size(), -1);
+  for(int targetIdx =0; targetIdx<target_p4.size(); targetIdx++){
+    if(pre_sel_target[targetIdx] == 0) continue;
+    int refIdxFound = FindMatching(target_p4[targetIdx], ref_p4, deltaR_thr);
+    if(pre_sel_ref[refIdxFound] != 0) {
+      targetIndices[targetIdx] = refIdxFound;
+    }
+  }
+
+  return targetIndices;
+}
+
+int FindMatching(const bool pre_sel_target, const RVecB& pre_sel_ref, const LorentzVectorM& target_p4,
+  const RVecLV& ref_p4, const float dR_thr)
+{
+  int matched = -1; // Only one target, so size is 1 and initialized with false
+  for(size_t ref_idx = 0; ref_idx < pre_sel_ref.size(); ref_idx++) {
+    if(pre_sel_ref[ref_idx] == 0) continue;
+    auto dR_current = ROOT::Math::VectorUtil::DeltaR(target_p4, ref_p4[ref_idx]);
+    if(dR_current < dR_thr && pre_sel_target) {
+      matched = ref_idx;
+    }
+  }
+  return matched;
```

Comment:
```suggestion
  double deltaR_min = deltaR_thr;
  int current_idx = -1;
  if(pre_sel_target) {
    for(int refIdx =0; refIdx<ref_p4.size(); refIdx++) {
      if(pre_sel_ref[refIdx]) {
        auto dR_targetRef= ROOT::Math::VectorUtil::DeltaR(target_p4, ref_p4.at(refIdx));
        if ( dR_targetRef < deltaR_min ) {
          deltaR_min = dR_targetRef ;
          current_idx = refIdx;
        }
      }
  }
  return current_idx;
```

Reply:
This doesn't seem resolved yet, because the current implementation doesn't select the closest object